addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -316,9 +303,63 @@ def _get_livechat_discuss_channel_vals(self, chatbot_script=None, agent=None):
             'chatbot_current_step_id': chatbot_script._get_welcome_steps()[-1].id if chatbot_script else False,
             "anonymous_name": visitor_user.display_name or guest.name,
             'channel_type': 'livechat',
-            'name': name,
+            'name': channel_name,
+        }
+
+    def _get_agent_member_vals(self, operator_info, last_interest_dt, now):
+        agent = operator_info['agent']
+        chatbot_script = operator_info['chatbot_script']
+        operator_partner = operator_info['operator_partner']
+        return {
+            "chatbot_script_id": chatbot_script.id if not agent else False,
+            "last_interest_dt": last_interest_dt,
+            "livechat_member_type": "agent" if agent else "bot",
+            "partner_id": operator_partner.id,
+            "unpin_dt": now,
         }
 
+    def _get_channel_name(self, operator_info, visitor_user=None, guest=None):
+        agent = operator_info['agent']
+        chatbot_script = operator_info['chatbot_script']
+
+        if chatbot_script:
+            channel_name = chatbot_script.title
+        else:
+            channel_name = ' '.join([
+                visitor_user.display_name if visitor_user else guest.name,
+                agent.livechat_username or agent.name
+            ])
+        return channel_name
+
+    def _get_operator_info(self, operator_lookup_params):
```

Comment:
I'm still not sure about `operator_lookup_params` in the route, but I'm sure in internal methods we want to spread the params instead, otherwise the API is unreadable and it's impossible to know what are the expected keys.
Base method can simply have * before the params to force keyword and **kwargs built-in to support overrides.